core/Archive/ArchiveInvalidator.php

```diff
@@ -590,12 +603,47 @@ public function removeInvalidationsSafely($idSites, $pluginName)
     {
         try {
             $this->removeInvalidations($idSites, $pluginName);
+            $this->removeInvalidationsFromDistributedList($idSites, $pluginName);
         } catch (\Throwable $ex) {
             $logger = StaticContainer::get(LoggerInterface::class);
             $logger->debug("Failed to remove invalidations the for $pluginName plugin.");
         }
     }
 
+    public function removeInvalidationsFromDistributedList($idSites, $pluginName = null)
+    {
+        $list = new ReArchiveList();
+        $entries = $list->getAll();
+
+        foreach ($entries as $index => $entry) {
+            $entry = @json_decode($entry, true);
+            if (empty($entry)) {
+                continue;
```

Comment:
wouldn't it make sense to always remove entries that can't be decoded? Otherwise they might kept in the list forever

Reply:
👍